filter.md

```diff
@@ -0,0 +1,217 @@
+
+These are the current severity levels:
+
+- `Tips` These are things that aren't necessarily wrong, but there may be a better, more idiomatic way to do it. This may also include performance issues.
+- `Untidy` This code smells. The player is unlikely to be impacted directly, but developers working on this codebase will likely experience maintenance headaches.
+- `Warning` This will result in glitches that will noticeably impact the player's gaming experience. Missing translations are an example.
+- `Error` This code probably doesn't work as intended. The player may experience bugs.
+- `Fatal` This is likely to cause crashes.
+
+You can add the severity trigger like so:
+
+- `severity == Tips` (Only match tips.)
+- `severity >= Warning` (Match Warnings or higher.)
+- `severity < Error` (Match anything below Error.)
+
+### Confidence
+
+Describes how confident Tiger is that a problem is real and not a false positive. If you are spammed with many false
+positives, you may try raising the required confidence level.
+
+These are the current confidence levels. They are subject to change.
+
+- `Weak` Tiger is not very confident about this report, it's quite likely to be a false positive.
+- `Reasonable` The default level.
+- `Strong` Tiger is very confident that this is a real problem.
+
+You can add the confidence trigger like so:
+
+- `confidence == Weak` (Only match weak.)
+- `confidence >= Reasonable` (Match Reasonable or higher.)
+- `confidence < Strong` (Match anything below Strong.)
+
+### Key
+
+Every report has a `key`, it describes the type of problem that is being reported about. You can find the key in
+parentheses on the first line of the report. For example, in the report below, the key is `missing-localization`:
+
+```
+Error(missing-localization): missing english localization key Badlay
+   --> [CK3] history/characters/afar.txt
+267 |  name = "Badlay"
+    |         ^ 
+```
+
+You can add a key trigger like so:
+
+- `key = missing-localization` <== This will match reports with the `missing-localization` key.
+
+### File
+
+You can target files and folders. Reports that mention the file will be matched by this trigger.
+
+Example:
+
+- `file = common/` This matches any report that mentions a file inside the `common/` directory.
+- `file = history/characters/afar.txt` This matches any report that mentions that specific file.
+
+# Migrating from `ignore` to `filter`
+
+Filtering was previously done through `ignore` blocks.
+
+Suppose this is our old setup:
+
+```
+ignore = {
+    key = duplicate-field
+}
+ignore = {
+    file = events/travel_events/travel_events.txt
+}
+ignore = {
+    key = duplicate-item
+    file = common/decisions/80_major_decisions.txt
+    file = common/script_values/00_basic_values.txt
+    file = common/defines/00_defines.txt
+}
+```
+
+We can replicate this like so:
+
+```
+trigger = {
+    NOR = {
+        key = duplicate-field
+        file = events/travel_events/travel_events.txt
+    }
+    NAND = {
```

Comment:
NAND is the hardest to understand so it's a bit of a problem if that's going to be the most common one

Reply:
I could add syntactic sugar for that one. Something like 
```
keys_in_files = {
    keys = {
         key1 key2 key3
    }
    files = {
         file2 file2 file3
    }
}
```

Reply:
That would be good :)